src/Microsoft.VisualStudio.Jdt.Tests/JsonTransformationTestLogger.cs

```diff
@@ -0,0 +1,132 @@
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Mock logger to test <see cref="JsonTransformation"/>
+    /// </summary>
+    public class JsonTransformationTestLogger : IJsonTransformationLogger
+    {
+        /// <summary>
+        /// Gets the error log
+        /// </summary>
+        public List<TestLogEntry> ErrorLog { get; } = new List<TestLogEntry>();
+
+        /// <summary>
+        /// Gets the warning log
+        /// </summary>
+        public List<TestLogEntry> WarningLog { get; } = new List<TestLogEntry>();
+
+        /// <summary>
+        /// Gets the message log
+        /// </summary>
+        public List<TestLogEntry> MessageLog { get; } = new List<TestLogEntry>();
+
+        /// <inheritdoc/>
+        public void LogError(string message)
+        {
+            this.ErrorLog.Add(new TestLogEntry(message, null, 0, 0, false));
+        }
+
+        /// <inheritdoc/>
+        public void LogError(string message, string fileName, int lineNumber, int linePosition)
+        {
+            this.ErrorLog.Add(new TestLogEntry(message, fileName, lineNumber, linePosition, false));
+        }
+
+        /// <inheritdoc/>
+        public void LogErrorFromException(Exception ex)
+        {
+            this.ErrorLog.Add(new TestLogEntry(ex.Message, null, 0, 0, true));
+        }
+
+        /// <inheritdoc/>
+        public void LogErrorFromException(Exception ex, string fileName, int lineNumber, int linePosition)
+        {
+            this.ErrorLog.Add(new TestLogEntry(ex.Message, fileName, lineNumber, linePosition, true));
+        }
+
+        /// <inheritdoc/>
+        public void LogMessage(string message)
+        {
+            this.MessageLog.Add(new TestLogEntry(message, null, 0, 0, false));
+        }
+
+        /// <inheritdoc/>
+        public void LogMessage(string message, string fileName, int lineNumber, int linePosition)
+        {
+            this.MessageLog.Add(new TestLogEntry(message, fileName, lineNumber, linePosition, false));
+        }
+
+        /// <inheritdoc/>
+        public void LogWarning(string message)
+        {
+            this.WarningLog.Add(new TestLogEntry(message, null, 0, 0, false));
+        }
+
+        /// <inheritdoc/>
+        public void LogWarning(string message, string fileName)
+        {
+            this.WarningLog.Add(new TestLogEntry(message, fileName, 0, 0, false));
+        }
+
+        /// <inheritdoc/>
+        public void LogWarning(string message, string fileName, int lineNumber, int linePosition)
+        {
+            this.WarningLog.Add(new TestLogEntry(message, fileName, lineNumber, linePosition, false));
+        }
+
+        /// <summary>
+        /// An test entry for the logger.
+        /// Corresponds to an error, warning or message
+        /// </summary>
+        public struct TestLogEntry
+        {
+            /// <summary>
+            /// The log message
+            /// </summary>
+            public string Message;
```

Comment:
for tests it doesn't matter much, but these should be properties with `{ get; set; }`. And then you can also remove the constructor and use property initializer.